### DISCUSSION TITLE
程序员可能必读书单推荐（一） - 面向信仰编程 · /books-1

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#649] [Announcements]
https://draveness.me/books-1/

**Comment by xie4ever** [comment]
资瓷，现在就去看

**Reply by jkrllezvous** [reply]
[好好学习](https://post.techwavelabs.org/redis/multi-thread-redis/)

**Comment by tanqiangyes** [comment]
DDIA神书

**Comment by chenmick** [comment]
DDIA 最近读了。对学习数据库，分布式什么的很有帮助，可以打通任督二脉 :)

**Comment by dawncold** [comment]
SICP刚被我卖掉，主要因为看不下去

**Comment by draveness** [comment]
> @dawncold
> SICP刚被我卖掉，主要因为看不下去

可以对着视频看

**Reply by Juminiy** [reply]
视频是哪里的

**Comment by woofyzhao** [comment]
SICP重点是每个练习题都要尽量做完。看了前4章很爽，第5章没空看了。

**Comment by szy0syz** [comment]
书架上有 SICP 和 DDIA，但只读了《🐗书》。

**Comment by draveness** [comment]
> 书架上有 SICP 和 DDIA，但只读了《🐗书》。

🐗书是啥

**Reply by ChangwangChen** [reply]
DDIA 书籍封面是头猪

**Comment by maxnoodles** [comment]
去年4月份，向大佬求书时推荐了 sicp 和 重构，2本都已看完，真的受益匪浅，sicp 流的理念深深震撼了我的编程思想。重构不说了，现在每一个函数都带着这本书的味道。这篇文章推荐的2本书已经纳入书单，非常非常感谢大佬！

**Comment by draveness** [comment]
> @maxnoodles 
> 去年4月份，向大佬求书时推荐了 sicp 和 重构，2本都已看完，真的受益匪浅，sicp 流的理念深深震撼了我的编程思想。重构不说了，现在每一个函数都带着这本书的味道。这篇文章推荐的2本书已经纳入书单，非常非常感谢大佬！

是嘛，这篇文章可能就是你当时问过之后我才想写的

**Comment by szy0syz** [comment]
> @draveness
> > 书架上有 SICP 和 DDIA，但只读了《🐗书》。
> 
> 🐗书是啥

就是DDIA 😂

**Comment by dawncold** [comment]
Youtube上搜到这个视频系列，是不是你说的？
https://www.youtube.com/channel/UC1wRNPq7rbaG-drFIVZdtxA
--
Zhen Tian

**Comment by OOp001** [comment]
DDIA有能力的话最好看原版，翻译版本有一些其实没有很准确的翻译

**Reply by wingor** [reply]
的确是的，有些地方读起来让人困惑，感觉跟翻译有关系

**Comment by ikingye** [comment]
《深入理解计算机系统》应该是第一梯队

**Comment by wangtuchuan** [comment]
> @maxnoodles
> 去年4月份，向大佬求书时推荐了 sicp 和 重构，2本都已看完，真的受益匪浅，sicp 流的理念深深震撼了我的编程思想。重构不说了，现在每一个函数都带着这本书的味道。这篇文章推荐的2本书已经纳入书单，非常非常感谢大佬！

大佬，问一下重构的全称是啥呀

**Comment by maxnoodles** [comment]
> @wang765256860
> > @maxnoodles
> > 去年4月份，向大佬求书时推荐了 sicp 和 重构，2本都已看完，真的受益匪浅，sicp 流的理念深深震撼了我的编程思想。重构不说了，现在每一个函数都带着这本书的味道。这篇文章推荐的2本书已经纳入书单，非常非常感谢大佬！
> 
> 大佬，问一下重构的全称是啥呀

重构 改善既有代码的设计 第2版

**Comment by yuanjize** [comment]
现代操作系统 第四版 挺好的

**Comment by fkdtz** [comment]
之前再推上留过几次言，这回大佬终于推荐书单了。「它却一开始就支持了函数式的编程范式」是不是拼写错误，应该是「它从一开始...」

**Comment by tangyiheng2021** [comment]
DDIA yyds

**Comment by ghost** [comment]
请教大佬您的网站构成呢：DNS在哪里购买？网站系统是哪款开源代码吗？是部署在阿里云吗？

**Comment by draveness** [comment]
> 请教大佬您的网站构成呢：DNS在哪里购买？网站系统是哪款开源代码吗？是部署在阿里云吗？

DNS 在 DNSPod 上的，网站系统整个是 Kubernetes 集群，在 Azure Kubernetes 服务商

**Comment by ghost** [comment]
> @draveness
> > 请教大佬您的网站构成呢：DNS在哪里购买？网站系统是哪款开源代码吗？是部署在阿里云吗？
> 
> DNS 在 DNSPod 上的，网站系统整个是 Kubernetes 集群，在 Azure Kubernetes 服务商

网站代码是用的开源软件吗？哪款开源软件呢？

**Comment by qiuhuiming** [comment]
马上下单

**Comment by planb137** [comment]
Draven, 有推荐linux操作系统的书籍吗？:smile:

**Comment by Darrenzzy** [comment]
> @planb137
> Draven, 有推荐linux操作系统的书籍吗？:smile:

推荐这本：Christian Benvenuti, 深入理解 LINUX 网络技术内幕，Understanding Linux Network Internals, 2005

go夜读68期有过一次对linux的分享，讲的不错。参考的书籍是这本。

**Comment by yexua** [comment]
你好，非常喜欢你的文章，可以在博客中加入图片放大插件吗？有些文章里的图片缩放导致观感不太好，需要在新标签页中打开图片

**Comment by qiuhuiming** [comment]
DDIA这本github有竟然有😁

**Comment by tabris233** [comment]
大佬 要不要给博客加一个搜索功能！ 能在所有文章检索关键字的那种。

有很多时候想回顾一下某些内容，但不太好找到是哪篇文章，光看标题有时候不太行。

**Comment by imageslr** [comment]
> @tabris233
> 大佬 要不要给博客加一个搜索功能！ 能在所有文章检索关键字的那种。
> 
> 有很多时候想回顾一下某些内容，但不太好找到是哪篇文章，光看标题有时候不太行。

作者的文章在 google 索引量很高，可以使用 google 搜索，配合 site: 条件

**Comment by imageslr** [comment]
> @yexua
> 你好，非常喜欢你的文章，可以在博客中加入图片放大插件吗？有些文章里的图片缩放导致观感不太好，需要在新标签页中打开图片

可以尝试装一个 chrome 插件就好啦，搜索 chrome image preview。

**Comment by kuzhao** [comment]
> @draveness
> > 请教大佬您的网站构成呢：DNS在哪里购买？网站系统是哪款开源代码吗？是部署在阿里云吗？
> 
> DNS 在 DNSPod 上的，网站系统整个是 Kubernetes 集群，在 Azure Kubernetes 服务商

单结点AKS的话不如container instance经济哦

**Comment by eddiearc** [comment]
DDIA NB

**Comment by ObserverWho** [comment]
DDIA githun有翻译版，准备拜读一下

**Comment by ChenHanZhang** [comment]
> @draveness
> > 书架上有 SICP 和 DDIA，但只读了《🐗书》。
> 
> 🐗书是啥

就是DDIA，封面是头野猪hhh

**Comment by asukaminato0721** [comment]
假如 SICP 刷不下去, 可以试一下 python 版的 cs61a. https://cs61a.org/ 会更新每年的题, 有 lab 可以做. 然后我把 2021 summer 的题基本做完了. https://github.com/wuyudi/cs61a-2021-summer

**Comment by jl4386** [comment]
大佬 什么时候更新啊

**Comment by pacey-li** [comment]
推荐周爱民的《程序原本》、《我的架构思想》https://github.com/aimingoo/my-ebooks
架构相关的还有一本《架构整洁之道》可以看看

**Comment by ancientmodern** [comment]
推荐几本我读过/正在读并且觉得不错的CS基础书籍（本科水平）
1. CSAPP，这个应该算公认的了
2. [计算机网络—自顶向下方法](http://gaia.cs.umass.edu/kurose_ross/index.php)，现在不管是前后端还是infra都和计算机网络高度相关，这本书讲得不深但可读性很强，用来入门很不错。作者更新速度也很快，已经出到第八版了，还有配套的视频
3. [OSTEP](https://pages.cs.wisc.edu/~remzi/OSTEP/)，内容和MO4这些老牌教科书比相对浅显，但原理和设计讲得很清楚，配套的作业也不错……强烈推荐用这本书入门OS
4. [Craftingin Ierpreters](https://craftinginterpreters.com/contents.html)，和大佬的go一样也是博客形式起家的，推荐的理由还是可读性强……龙书虎书太难顶了

**Reply by draveness** [reply]
最后一本没看过，其他都看过

**Comment by hooklab** [comment]
csapp 入行必读

**Comment by jkrllezvous** [comment]
收藏了, [好好学习](https://post.techwavelabs.org/redis/multi-thread-redis/)
